crates/typst/src/diag.rs

```diff
@@ -161,6 +161,18 @@ impl SourceDiagnostic {
         }
     }
 
+    /// Create a new error with hints.
+    pub fn hinted_error(span: Span, hinted_message: HintedString) -> Self {
```

Comment:
I think it's nicer if we keep this in the `At` impl, so that there is only one way to do this for API consumers.